esinstall/src/entrypoints.ts

```diff
@@ -218,20 +214,28 @@ export function resolveEntrypoint(
     packageLookupFields,
   });
 
-  // Some packages are types-only. If this is one of those packages, resolve with that.
-  if (!foundEntrypoint && hasTypes(depManifest)) {
-    const typesLoc = (depManifest.types || depManifest.typings) as string;
-    return path.join(depManifestLoc, '..', typesLoc);
-  }
   // Sometimes packages don't give an entrypoint, assuming you'll fall back to "index.js".
   if (!foundEntrypoint) {
-    foundEntrypoint = 'index.js';
+    for(let possibleEntrypoint of ['index.js', 'index.json']) {
+      try {
+        return realpathSync.native(resolve.sync(path.join(depManifestLoc || '', '..', possibleEntrypoint)));
+      } catch {
+
```

Comment:
move the throw error below into the catch block? Since its only hit on catch, I think that makes sense.

Reply:
This catch is inside of the loop. We only want to throw at the completion of the loop after checking the possible implicit mains (index.js and index.json). So we do the loop and at that point know we don't have an entrypoint we can find, so throw there.

Reply:
This catch only exists because `try { something() }` isn't valid JS.